### PR TITLE
Support generating videos in `eval_policy`

### PR DESCRIPTION
**Pull request by AdamGleave**
Adds support for generating video renderings of policies acting in environments with `eval_policy`.

I include a new `VideoWrapper` class (adapted from https://github.com/HumanCompatibleAI/adversarial-policies/blob/master/src/aprl/policies/wrappers.py). We could instead use `gym.Monitor` (not to be confused with `stable_baselines.bench.Monitor`, which we already use) but it (a) has a lot of features we don't need, and (b) always makes a new video file for each episode, which I find annoying (all episodes in one MP4 is something I want more often and I am too lazy to concatenate).

I wanted to test this, so I added a dummy X server in our Dockerfile, and then also enabled testing for rendering. I've pushed the new images to DockerHub.

### CODE REVIEW

**Comment by AdamGleave** [comment]
Before merging this PR: get https://github.com/HumanCompatibleAI/seals/pull/44 merged and make new release then update `setup.py` here to point to new release. We're using some of `seals` new render testing functionality. (I could split this into a separate PR if desired.)

**Comment by shwang** [comment]
Missing some diff coverage in VideoWrapper for `if not self.single_video:`, `VideoWrapper.close` (comes for free if `not self.single_video`).

**Review by shwang — CHANGES_REQUESTED** [review]
Overall looks good; thanks for adding this in. Some minor comments, and would be nice for have a test for `VideoWrapper.single_video = False` so we can cover the entire diff.

File: src/imitation/envs/examples/model_envs.py
Comment (AdamGleave) [reply src/imitation/envs/examples/model_envs.py:70]: Test suite errors because we added to `seals` a check for `dtype`. This is a consequence of upgrading the `seals` version to the GitHub master not rendering per se. I could split this into a separate PR if desired but it only affects a couple of places.

File: src/imitation/scripts/eval_policy.py
Comment (AdamGleave) [reply src/imitation/scripts/eval_policy.py]: hmm, I actually think I'm abusing terminology here, and it should be a "higher-order function". I wanted to call it `video_wrapper` but that clashed with the file of the same name. I've renamed to `video_wrapper_factory`.

File: src/imitation/util/util.py
Comment (AdamGleave) [reply src/imitation/util/util.py:44]: done, good catch

**Review by AdamGleave — COMMENTED** [review]
Thanks for the review! Agree with your comments, believe I've resolved everything. I've also added a test case for `single_videos` to improve coverage.

File: src/imitation/scripts/eval_policy.py
Comment (shwang) [inline src/imitation/scripts/eval_policy.py:88]:
```suggestion
      render_fps: The target number of frames per second to render on screen.
```

File: src/imitation/scripts/eval_policy.py
Comment (shwang) [inline src/imitation/scripts/eval_policy.py:90]:
```suggestion
      video_kwargs: Keyword arguments passed through to `video_wrapper.VideoWrapper`.
```

File: src/imitation/util/util.py
Comment (shwang) [inline src/imitation/util/util.py:61]:
```suggestion
            of the wrappers specified in the sequence. The argument should be a Callable
            accepting two arguments, the Env to be wrapped and the environment index.
```

File: src/imitation/util/video_wrapper.py
Comment (shwang) [inline src/imitation/util/video_wrapper.py:15]:
```suggestion
        self, env: gym.Env, directory: types.AnyPath, single_video: bool = True,
```

File: tests/test_scripts.py
Comment (shwang) [inline tests/test_scripts.py:81]: Oh, looks like we should actually be using `single_video=False` to increase coverage, actually.
```suggestion
    {"videos": True, "video_kwargs": {"single_video": False}},
```

**Review by shwang — APPROVED** [review]
Few docstring nits. More significantly we still don't have diff coverage because I suggested the oppositely-argumented test case `single_video=True`.

Don't think this should require another review after we have diff coverage.

**Review by AdamGleave — COMMENTED** [review]
Thanks for the review and catching my dumb typos and `single_video=True` rather than `False`. Diff coverage is fixed now. 

I made some changes to CircleCI config to increase size of executor since was running into OOM error. Oddly it's now running slower than before, despite more total CPUs (1 instance with 8 CPUs in parallel, compared to 3 instances with 2 CPUs). I dunno what's causing this, will try and dig into it tomorrow, but would welcome any ideas if anyone's seen this before.

**Comment by AdamGleave** [comment]
Hmm, running the tests locally even with `-n 2` on a 16 core machine I'm seeing 100% CPU saturation at some points. So I suspect that each parallel instance of PyTorch may be trying to use all the available CPUs. There's probably some way to turn this off.

**Comment by shwang** [comment]
I’ve had similar problems with parallelizing local tests on my workstation.

**Comment by AdamGleave** [comment]
Works well now (test suite running in <3m) after I set PyTorch and JAX to only run single-threaded, so I think that was the main issue. We could probably get more speedups by tweaking settings slightly, but this is good enough for me. Merging.

**Pull request merged**
